packages/node/src/algorand/api.algorand.ts

```diff
@@ -117,15 +118,40 @@ export class AlgorandApi {
         this.getHeaderOnly(blockHeight),
         this.paginatedTransactions(blockHeight),
       ]);
-      const header = camelCaseObjectKey(blockHeader);
-      header.transactions = camelCaseObjectKey(paginatedTransactionsResults);
-      return header;
+
+      return this.constructBlock({
+        ...camelCaseObjectKey(blockHeader),
+        transactions: camelCaseObjectKey(paginatedTransactionsResults),
+      });
     } catch (e) {
       logger.error(`Failed to paginate round ${blockHeight}`);
       throw e;
     }
   }
 
+  private constructBlock(block: AlgorandBlock): AlgorandBlock {
+    const newBlock = {
+      ...block,
+      getTransactionsByGroup: (groupId: string) =>
+        transactions.filter((tx) => tx.group === groupId),
+      toJSON(): string {
+        return JSON.stringify(omit(this, ['getTransactionsByGroup', 'toJSON']));
```

Comment:
would `this` be in the context of block or the `AlgorandApi` class ?

Reply:
`this` is correct. I've updated the test and found another bug